gateway/src/main/java/io/zeebe/gateway/EndpointManager.java

```diff
@@ -320,13 +322,22 @@ public void updateJobRetries(
       return;
     }
 
+    suppressCancelledException(grpcRequest, streamObserver);
     brokerClient.sendRequest(
         brokerRequest,
         (key, response) -> consumeResponse(responseMapper, streamObserver, key, response),
         error -> streamObserver.onError(convertThrowable(error)),
         timeout);
   }
 
+  private <GrpcRequestT, GrpcResponseT> void suppressCancelledException(
+      final GrpcRequestT grpcRequest, final StreamObserver<GrpcResponseT> streamObserver) {
+    final ServerCallStreamObserver<GrpcResponseT> serverObserver =
+        (ServerCallStreamObserver<GrpcResponseT>) streamObserver;
+    serverObserver.setOnCancelHandler(
```

Comment:
What happend before without the handler again? Sorry :sweat_smile:

Reply:
Because these exceptions are fixed with the addition check right?

```
I 2020-01-30T03:17:21.480895200Z io.grpc.StatusRuntimeException: CANCELLED: call already cancelled
	at io.grpc.Status.asRuntimeException(Status.java:524) ~[grpc-api-1.26.0.jar:1.26.0]
	at io.grpc.stub.ServerCalls$ServerCallStreamObserverImpl.onCompleted(ServerCalls.java:368) ~[grpc-stub-1.26.0.jar:1.26.0]
	at io.zeebe.gateway.impl.job.LongPollingActivateJobsRequest.complete(LongPollingActivateJobsRequest.java:73) ~[zeebe-gateway-0.23.0-SNAPSHOT.jar:0.23.0-SNAPSHOT]
	at io.zeebe.gateway.impl.job.LongPollingActivateJobsRequest.timeout(LongPollingActivateJobsRequest.java:95) ~[zeebe-gateway-0.23.0-SNAPSHOT.jar:0.23.0-SNAPSHOT]
	at io.zeebe.gateway.impl.job.LongPollingActivateJobsHandler.lambda$addTimeOut$9(LongPollingActivateJobsHandler.java:191) ~[zeebe-gateway-0.23.0-SNAPSHOT.jar:0.23.0-SNAPSHOT]
	at io.zeebe.util.sched.ActorJob.invoke(ActorJob.java:76) [zeebe-util-0.23.0-SNAPSHOT.jar:0.23.0-SNAPSHOT]
	at io.zeebe.util.sched.ActorJob.execute(ActorJob.java:39) [zeebe-util-0.23.0-SNAPSHOT.jar:0.23.0-SNAPSHOT]
	at io.zeebe.util.sched.ActorTask.execute(ActorTask.java:115) [zeebe-util-0.23.0-SNAPSHOT.jar:0.23.0-SNAPSHOT]
	at io.zeebe.util.sched.ActorThread.executeCurrentTask(ActorThread.java:107) [zeebe-util-0.23.0-SNAPSHOT.jar:0.23.0-SNAPSHOT]
	at io.zeebe.util.sched.ActorThread.doWork(ActorThread.java:91) [zeebe-util-0.23.0-SNAPSHOT.jar:0.23.0-SNAPSHOT]
	at io.zeebe.util.sched.ActorThread.run(ActorThread.java:195) [zeebe-util-0.23.0-SNAPSHOT.jar:0.23.0-SNAPSHOT]
```

Reply:
Yes, it's the current way grpc-java has to disable throwing an exception when the call was already cancelled (not ideal imo, but that's their current recommended workaround)